src/crypto/ed25519.rs

```diff
@@ -0,0 +1,240 @@
+use crate::error::CryptoError;
+use ed25519_dalek::{Signer, SigningKey, Verifier, VerifyingKey};
+use rand::rngs::OsRng;
+use wasm_bindgen::prelude::*;
+
+/// Generate a new Ed25519 signing key using secure random number generation.
+/// Returns 32 bytes of raw key material suitable for use with other Ed25519 functions.
+#[wasm_bindgen]
+pub fn new_ed25519_signing_key() -> Vec<u8> {
```

Comment:
It should be faster to return a boxed slice (`Box<[u8]>`) since the return value never needs to change size. Please may you do that for all functions where that is true?

To do that you should be able to just declare the return type as `Box<[u8]>` and then do `signing_key.to_bytes().into()`, for example

Reply:
Yup, that makes sense!